### PR TITLE
Emby.Server.Implementations code cleanup and warning fixes (Part 3)

### PR DESCRIPTION
**Pull request by barronpm**
**Changes**
Documents and cleans up code in Emby.Server.Implementations.Data.

This is the third part of a series of PR's that hopefully will eliminate all warnings in the Emby.Server.Implementations project.

**Related Issues**
#2149

### CODE REVIEW

**Comment by barronpm** [comment]
There is another warning on line 45 of UdpServerEntryPoint.cs, but I'm not sure how to fix it. The warning is about the `RunAsync` method, which is declared async but runs synchronously. The problem is that it is inherited from the `IServerEntryPoint` interface, so the method signature can't be changed.

File: Emby.Server.Implementations/Images/BaseDynamicImageProvider.cs [hunk 1]
```diff
@@ -31,26 +40,51 @@ protected BaseDynamicImageProvider(IFileSystem fileSystem, IProviderManager prov
...
+        /// </summary>
         protected IProviderManager ProviderManager { get; }
 
+        /// <summary>
+        /// Gets the application paths.
+        /// </summary>
         protected IApplicationPaths ApplicationPaths { get; }
 
+        /// <summary>
+        /// Gets and sets the image processor.
```
Comment (mark-monteiro) [inline Emby.Server.Implementations/Images/BaseDynamicImageProvider.cs:59]:
```suggestion
        /// Gets or sets the image processor.
```

File: Emby.Server.Implementations/Images/BaseDynamicImageProvider.cs [hunk 2]
```diff
@@ -75,6 +109,14 @@ public async Task<ItemUpdateType> FetchAsync(T item, MetadataRefreshOptions opti
...
         }
 
+        /// <summary>
+        /// Fetches the
+        /// </summary>
+        /// <param name="item">The item.</param>
+        /// <param name="imageType">The image type.</param>
+        /// <param name="options">The metadata refresh options.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>A task </returns>
```
Comment (mark-monteiro) [inline Emby.Server.Implementations/Images/BaseDynamicImageProvider.cs:119]:
```suggestion
        /// <summary>
        /// Fetches the image asynchronously.
        /// </summary>
        /// <param name="item">The item.</param>
        /// <param name="imageType">The image type.</param>
        /// <param name="options">The metadata refresh options.</param>
        /// <param name="cancellationToken">The cancellation token.</param>
        /// <returns>A <see cref="Task"/> representing the fetch operation.</returns>
```

File: Emby.Server.Implementations/Images/BaseDynamicImageProvider.cs [hunk 3]
```diff
@@ -124,13 +174,31 @@ protected Task<ItemUpdateType> FetchAsync(BaseItem item, ImageType imageType, Me
...
+        /// <returns>A read-only list containing items with images.</returns>
         protected abstract IReadOnlyList<BaseItem> GetItemsWithImages(BaseItem item);
 
+        /// <summary>
+        /// Creates a thumbnail collage.
+        /// </summary>
+        /// <param name="primaryItem">The primary item.</param>
+        /// <param name="items">The items.</param>
+        /// <param name="outputPath">The output path.</param>
+        /// <returns>The resulting path, or null.</returns>
```
Comment (mark-monteiro) [inline Emby.Server.Implementations/Images/BaseDynamicImageProvider.cs:190]:
```suggestion
        /// <returns>The resulting path, or null if the collage was not created.</returns>
```

File: Emby.Server.Implementations/Images/BaseDynamicImageProvider.cs [hunk 4]
```diff
@@ -153,16 +221,39 @@ protected virtual IEnumerable<string> GetStripCollageImagePaths(BaseItem primary
                 .Where(i => !string.IsNullOrEmpty(i));
         }
 
+        /// <summary>
+        /// Creates a poster collage.
+        /// </summary>
+        /// <param name="primaryItem">The primary item.</param>
+        /// <param name="items">The items.</param>
+        /// <param name="outputPath">The output path.</param>
+        /// <returns>The resulting path, or null.</returns>
```
Comment (mark-monteiro) [inline Emby.Server.Implementations/Images/BaseDynamicImageProvider.cs:230]:
```suggestion
        /// <returns>The resulting path, or null if the collage was not created.</returns>
```

File: Emby.Server.Implementations/Images/BaseDynamicImageProvider.cs [hunk 5]
```diff
@@ -153,16 +221,39 @@ protected virtual IEnumerable<string> GetStripCollageImagePaths(BaseItem primary
...
             return CreateCollage(primaryItem, items, outputPath, 400, 600);
         }
 
+        /// <summary>
+        /// Creates a square collage.
+        /// </summary>
+        /// <param name="primaryItem">The primary item.</param>
+        /// <param name="items">The items.</param>
+        /// <param name="outputPath">The output path.</param>
+        /// <returns>The resulting path, or null.</returns>
```
Comment (mark-monteiro) [inline Emby.Server.Implementations/Images/BaseDynamicImageProvider.cs:242]:
```suggestion
        /// <returns>The resulting path, or null if the collage was not created.</returns>
```

File: Emby.Server.Implementations/Images/BaseDynamicImageProvider.cs [hunk 6]
```diff
@@ -153,16 +221,39 @@ protected virtual IEnumerable<string> GetStripCollageImagePaths(BaseItem primary
...
 
+        /// <summary>
+        /// Creates a thumbnail collage.
+        /// </summary>
+        /// <param name="primaryItem">The primary item.</param>
+        /// <param name="items">The items.</param>
+        /// <param name="outputPath">The output path.</param>
+        /// <param name="width">The width.</param>
+        /// <param name="height">The height.</param>
+        /// <returns>The resulting path, or null.</returns>
```
Comment (mark-monteiro) [inline Emby.Server.Implementations/Images/BaseDynamicImageProvider.cs:256]:
```suggestion
        /// <returns>The resulting path, or null if the collage could not be created.</returns>
```

File: Emby.Server.Implementations/Images/BaseDynamicImageProvider.cs [hunk 7]
```diff
@@ -194,6 +285,16 @@ private string CreateCollage(BaseItem primaryItem, IEnumerable<BaseItem> items,
...
 
+        /// <summary>
+        /// Creates an image.
+        /// </summary>
+        /// <param name="item">The item.</param>
+        /// <param name="itemsWithImages">The items with the images.</param>
+        /// <param name="outputPathWithoutExtension">The output path without an extension.</param>
+        /// <param name="imageType">The image type.</param>
+        /// <param name="imageIndex">The image index.</param>
+        /// <returns>The resulting image's path, or null.</returns>
```
Comment (mark-monteiro) [inline Emby.Server.Implementations/Images/BaseDynamicImageProvider.cs:296]:
```suggestion
        /// <returns>The resulting image's path, or null if <paramref name="itemsWithImages"/> is empty.</returns>
```

File: Emby.Server.Implementations/Images/BaseDynamicImageProvider.cs [hunk 8]
```diff
@@ -244,37 +346,45 @@ public bool HasChanged(BaseItem item, IDirectoryService directoryServicee)
...
             return age.TotalDays > MaxImageAgeDays;
         }
 
+        /// <summary>
+        /// Creates an image based on the provided items at the specified path.
+        /// </summary>
+        /// <param name="itemsWithImages">The items.</param>
+        /// <param name="outputPathWithoutExtension">The output path without the file extension.</param>
+        /// <param name="imageType">The image type.</param>
+        /// <returns>The output path of the created image, or null.</returns>
```
Comment (mark-monteiro) [inline Emby.Server.Implementations/Images/BaseDynamicImageProvider.cs:387]:
```suggestion
        /// <returns>The output path of the created image, or null if the image was not created.</returns>
```

File: Emby.Server.Implementations/Localization/LocalizationManager.cs [hunk 9]
```diff
@@ -73,31 +73,29 @@ public async Task LoadAll()
                 string countryCode = resource.Substring(RatingsResource.Length, 2);
                 var dict = new Dictionary<string, ParentalRating>(StringComparer.OrdinalIgnoreCase);
 
-                using (var str = _assembly.GetManifestResourceStream(resource))
-                using (var reader = new StreamReader(str))
+                await using var str = _assembly.GetManifestResourceStream(resource);
```
Comment (mark-monteiro) [inline Emby.Server.Implementations/Localization/LocalizationManager.cs:76]: Still need to call `.configureAwait(false)` for `await using`. Unfortunately the syntax to do so isn't very nice :/

```suggestion
                var str = _assembly.GetManifestResourceStream(resource);
                await using var asyncDisposableStr = str.configureAwait(false)
```

File: Emby.Server.Implementations/Localization/LocalizationManager.cs [hunk 10]
```diff
@@ -119,52 +117,42 @@ private async Task LoadCultures()
 
             const string ResourcePath = "Emby.Server.Implementations.Localization.iso6392.txt";
 
-            using (var stream = _assembly.GetManifestResourceStream(ResourcePath))
-            using (var reader = new StreamReader(stream))
+            await using var stream = _assembly.GetManifestResourceStream(ResourcePath);
```
Comment (mark-monteiro) [inline Emby.Server.Implementations/Localization/LocalizationManager.cs:120]: Needs `ConfigureAwait(false)`

File: Emby.Server.Implementations/Localization/LocalizationManager.cs [hunk 11]
```diff
@@ -342,23 +330,21 @@ public string GetLocalizedString(string phrase, string culture)
 
         private async Task CopyInto(IDictionary<string, string> dictionary, string resourcePath)
         {
-            using (var stream = _assembly.GetManifestResourceStream(resourcePath))
+            await using var stream = _assembly.GetManifestResourceStream(resourcePath);
```
Comment (mark-monteiro) [inline Emby.Server.Implementations/Localization/LocalizationManager.cs:333]: Needs `ConfigureAwait(false)`

**Pull request closed without merging**